test/test_linearizer.py

```diff
@@ -628,22 +628,26 @@ def _assert_grouped_dims(prefix, dims, max_sizes, reverse_dims, expected_sizes):
     _assert_grouped_dims("gidx", (2,3), (16,16,16,), False, [2,3,1])
     _assert_grouped_dims("gidx", (2,3), (16,16,16,), True, [3,2,1])
     _assert_grouped_dims("gidx", (2,3,4,), (16,16,16,), False, [2,3,4])
-    _assert_grouped_dims("gidx", (2,3,4,5,), (16,16,16,), False, [6,4,5])
-    # _assert_grouped_dims("gidx", (2,3,4,5,), (16,16,16,), True, [5,12,2]) # this is the new linearizer way
-    # _assert_grouped_dims("gidx", (2,3,4,5,), (32,16,16,), True, [20,3,2]) # this is the new linearizer way
-    _assert_grouped_dims("gidx", (2,3,4,5,), (16,16,16,), True, [5,4,6]) # this is the old linearizer way - TODO: remove
+    _assert_grouped_dims("gidx", (2,3,4,5,), (16,16,16,), False, [6,4,5]) # collapse on onto the left most axis
+    _assert_grouped_dims("gidx", (2,3,4,5,), (4,16,16,), False, [2,12,5]) # collapse on left-most available axis (the left most is too small)
+    _assert_grouped_dims("gidx", (2,3,4,5,), (32,16,16,), True, [20,3,2]) # collapse on onto the left most axis
+    _assert_grouped_dims("gidx", (2,3,4,5,), (16,16,16,), True, [5,12,2]) # collapse on left-most available axis (the left most is too small)
     _assert_grouped_dims("gidx", (64,3,4,), (16,16,16,), False, [16,12,4])
     _assert_grouped_dims("gidx", (64,3,4,), (16,4,16,), False, [16,4,12])
     _assert_grouped_dims("gidx", (64,3,4,), (16,16,16,), True, [12,16,4])
+    _assert_grouped_dims("gidx", (128,3,4,), (16,4,256,), False, [16,4,24])
 
     with self.assertRaises(AssertionError): # dim too large and not factorable
       get_grouped_dims("gidx", 0, (23,), (16,16,16,), False,)
 
+    with self.assertRaises(AssertionError): # dim too large and not factorable
+      get_grouped_dims("gidx", 0, (128,3,4), (16,4,23,), False,)
+
     with self.assertRaises(AssertionError): # too large for sizes
       get_grouped_dims("gidx", 0, (2,3,4,5,6), (16,16,16,), False,)
 
     with self.assertRaises(AssertionError): # variable too large
-      get_grouped_dims("gidx", 0, (Variable("start_pos", 0, 17),3,4), (16,16,16,), False,)
+      get_grouped_dims("gidx", 0, (Variable("start_pos", 0, 16),3,4), (16,16,16,), False,)
```

Comment:
can you add some cases that has Variable and collapsed?

Reply:
oh i see `TODO: support sint properly`, maybe it's not supported yet?

Reply:
Oh right it's not supported yet, but I added a test case that passes and one that should when it is.